### ISSUE TITLE
An error occurred while Style/EvalWithLocation - undefined method `str_type?' for nil:NilClass

### ISSUE THREAD

**Issue by natematykiewicz** [rubocop-hq/rubocop#9441]
An error occurred while Style/EvalWithLocation cop was inspecting Gemfile:3:0.

--------

## Expected behavior

Run `rubocop` rubocop without error.

## Actual behavior

Running `rubocop` prints `An error occurred while Style/EvalWithLocation cop was inspecting Gemfile:3:0.`

Backtrace:
```
undefined method `str_type?' for nil:NilClass
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/style/eval_with_location.rb:67:in `on_send'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:136:in `public_send'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:136:in `block (2 levels) in trigger_restricted_cops'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:166:in `with_cop_error_handling'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:135:in `block in trigger_restricted_cops'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:134:in `each'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:134:in `trigger_restricted_cops'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:154:in `on_block'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:137:in `each'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:20:in `walk'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/team.rb:157:in `investigate_partial'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cop/team.rb:83:in `investigate'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:315:in `inspect_file'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:286:in `loop'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:132:in `block in file_offenses'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:157:in `file_offense_cache'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:131:in `file_offenses'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:122:in `process_file'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:101:in `block in each_inspected_file'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:100:in `each'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:100:in `reduce'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:100:in `each_inspected_file'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:86:in `inspect_files'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/runner.rb:47:in `run'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cli/command.rb:11:in `run'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cli/environment.rb:18:in `run'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cli.rb:65:in `run_command'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cli.rb:72:in `execute_runners'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/lib/rubocop/cli.rb:41:in `run'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/exe/rubocop:13:in `block in <top (required)>'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/opt/rbenv/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rubocop-1.9.0/exe/rubocop:12:in `<top (required)>'
/opt/rbenv/rbenv/versions/2.5.1/bin/rubocop:23:in `load'
```

## Steps to reproduce the problem

We add a custom method to our Gemfile as a helper.

Add this into your Gemfile:

```ruby
# frozen_string_literal: true

Bundler::Dsl.class_eval do
end
```

Run `rubocop -d`. See it error on the `class_eval` line.

## RuboCop version

```
$ bundle exec rubocop -V
1.9.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 2.5.1 x86_64-linux)
  - rubocop-performance 1.9.2
  - rubocop-rails 2.9.1
  - rubocop-rspec 2.1.0
```

**Comment by natematykiewicz** [comment]
I imagine #9440 is related (or the same thing). Not sure. The message is different, so I made my own issue.

**Comment by dvandersluis** [comment]
Thanks, this is a duplicate of #9433 and already fixed by #9435.

**Comment by natematykiewicz** [comment]
Excellent. Thanks @dvandersluis!
